File: srxproperty/src/dataObject/OnlineCommunicationPO.js
import { ObjectUtil, CommonUtil, DebugUtil } from "../utils";

class OnlineCommunicationPO {
  constructor(data) {
    if (!ObjectUtil.isEmpty(data)) {
      this.content = data.content;
      this.contentShort = data.contentShort;
      this.datePosted = data.datePosted;
      this.encryptedId = data.encryptedId;
      this.imageUrl = data.imageUrl;
      this.imageUrlSmall = data.imageUrlSmall;
      this.slug = data.slug;
      this.source = data.source;
      this.title = data.title;
    }
  }

  getImageUrl = () => {
    if (!ObjectUtil.isEmpty(this.imageUrl)) {
      return CommonUtil.handleImageUrl(this.imageUrl);
    }
    return "";
  };

  getImageSmallUrl = () => {
    if (!ObjectUtil.isEmpty(this.imageUrlSmall)) {
      return CommonUtil.handleImageUrl(this.imageUrlSmall);
    }
    return "";
  };

  getNewsURL = () => {
    const newsId = this.getNewsId();
    if (!ObjectUtil.isEmpty(newsId)) {
      const url = DebugUtil.retrieveStoreDomainURL();
      return url + "/singapore-property-news/" + newsId + "/" + this.slug;
    }
    return null;
  };

  //encryptedId d5c2t7v5o2b to 52752
  getNewsId = () => {
    const { encryptedId } = this;
    if (encryptedId) {
      const result = encryptedId.match(/\d/g).join("");
      return result;
    }
    return null;
  };
}

export { OnlineCommunicationPO };

File: agent-connect-android/app/src/main/java/sg/searchhouse/agentconnect/model/api/propertynews/OnlineCommunicationPO.kt
package sg.searchhouse.agentconnect.model.api.propertynews

import android.content.Context
import com.google.gson.annotations.SerializedName
import sg.searchhouse.agentconnect.util.ApiUtil

data class OnlineCommunicationPO(
    @SerializedName("id")
    val id: String,
    @SerializedName("title")
    val title: String,
    @SerializedName("content")
    val content: String,
    @SerializedName("contentShort")
    val contentShort: String,
    @SerializedName("source")
    val source: String,
    @SerializedName("url")
    val url: String,
    @SerializedName("slug")
    val slug: String,
    @SerializedName("encryptedId")
    val encryptedId: String,
    @SerializedName("imageUrl")
    val imageUrl: String,
    @SerializedName("imageUrlSmall")
    val imageUrlSmall: String,
    @SerializedName("datePosted")
    val datePosted: String
) {
    fun getFormattedContent(): String {
        return "<style>img{display: inline;height: auto;max-width: 100%;}p{line-height:1.6}div{line-height:1.6}</style>$content"
    }

    fun getPropertyNewsUrl(context: Context): String {
        return "${ApiUtil.getBaseUrl(context)}/singapore-property-news/$id/$slug"
    }
}